src/cpp/src/image_generation/stable_diffusion_3_pipeline.hpp

```diff
@@ -125,40 +125,47 @@ class StableDiffusion3Pipeline : public DiffusionPipeline {
 
         set_scheduler(Scheduler::from_config(root_dir / "scheduler/scheduler_config.json"));
 
+        // Temporary fix for GPU
+        ov::AnyMap updated_properties = properties;
+        if (device.find("GPU") != std::string::npos &&
+            updated_properties.find("INFERENCE_PRECISION_HINT") == updated_properties.end()) {
+            updated_properties["INFERENCE_PRECISION_HINT"] = ov::element::f32;
+        }
+
         const std::string text_encoder = data["text_encoder"][1].get<std::string>();
         if (text_encoder == "CLIPTextModelWithProjection") {
             m_clip_text_encoder_1 =
-                std::make_shared<CLIPTextModelWithProjection>(root_dir / "text_encoder", device, properties);
+                std::make_shared<CLIPTextModelWithProjection>(root_dir / "text_encoder", device, updated_properties);
         } else {
             OPENVINO_THROW("Unsupported '", text_encoder, "' text encoder type");
         }
 
         const std::string text_encoder_2 = data["text_encoder_2"][1].get<std::string>();
         if (text_encoder_2 == "CLIPTextModelWithProjection") {
             m_clip_text_encoder_2 =
-                std::make_shared<CLIPTextModelWithProjection>(root_dir / "text_encoder_2", device, properties);
+                std::make_shared<CLIPTextModelWithProjection>(root_dir / "text_encoder_2", device, updated_properties);
         } else {
             OPENVINO_THROW("Unsupported '", text_encoder_2, "' text encoder type");
         }
 
         const std::string text_encoder_3 = data["text_encoder_3"][1].get<std::string>();
         if (text_encoder_3 == "T5EncoderModel") {
-            m_t5_text_encoder = std::make_shared<T5EncoderModel>(root_dir / "text_encoder_3", device, properties);
+            m_t5_text_encoder = std::make_shared<T5EncoderModel>(root_dir / "text_encoder_3", device, updated_properties);
         }
 
         const std::string transformer = data["transformer"][1].get<std::string>();
         if (transformer == "SD3Transformer2DModel") {
-            m_transformer = std::make_shared<SD3Transformer2DModel>(root_dir / "transformer", device, properties);
+            m_transformer = std::make_shared<SD3Transformer2DModel>(root_dir / "transformer", device, updated_properties);
```

Comment:
Doesn't transformer work with fp16 precision? I think it was OK after the fix here https://github.com/openvinotoolkit/openvino/pull/27827

Reply:
reverted